### PR TITLE
Add `--ignore ID_REGEX` option to `dandi validate`

### LINKED ISSUE #1232: validate: add --ignore REGEX to ignore some errors
REGEX for ids, in light of https://github.com/dandi/dandi-cli/issues/1211 , so people on not yet organized files could use `dandi validate --ignore DANDI.NO_DANDISET_FOUND PATHs`

### PR DESCRIPTION
**Pull request by jwodder**
Closes #1232.

I also cleaned up some of the code in `validate.py` because I couldn't stop myself.

### CODE REVIEW

**Review by TheChymera — APPROVED** [review]
Looks good.

File: dandi/cli/cmd_validate.py [hunk 1]
```diff
@@ -111,50 +117,44 @@ def validate(
...
         for purview in purviews:
-            applies_to = [
-                i for i in issues if purview in [i.path, i.path_regex, i.dataset_path]
-            ]
+            applies_to = [i for i in issues if purview == i.purview]
             display_errors(
                 [purview],
                 [i.id for i in applies_to],
-                [i.severity for i in applies_to],
+                cast(List[Severity], [i.severity for i in applies_to]),
```
Comment (yarikoptic) [inline dandi/cli/cmd_validate.py:143]: me is wondering why this casting is needed since it seems it is explicit that it would be a list of Severities.

**Review by yarikoptic — APPROVED** [review]
Didn't have a chance to try but looks good to me. Let's proceed!

File: dandi/cli/cmd_validate.py (hunk 1 above)
Comment (jwodder) [reply dandi/cli/cmd_validate.py:143]: `severity` can be `None`, so mypy sees `[i.severity for i in applies_to]` as a `list[Optional[Severity]]`, yet we've filtered out the `None` severities above.

**Comment by yarikoptic** [comment]
Dear @jwodder - please resolve conflicts to your liking (I merged the Path tuneup in tests) and merge as long as it is green!

**Comment by jwodder** [comment]
FYI, the release action failed because of the E2BIG bug.  The only step left to do manually was build & upload to PyPI, which I took care of.

**Comment by yarikoptic** [comment]
eh - we waited too long for a release I guess. Thanks for taking care about it.

**Pull request merged**